library-sync.sh

```diff
@@ -3,6 +3,71 @@
 # utility script to gather template/imagestream content from https://github.com/openshift/library
 # and store it in this repo (cannot access other repos with dist git, and advised against git submodules
 
+# Without any commandline arguments, this script only updates the OCP supported samples. This behavior
+# can be modified using these commandline arguments:
+# * --okd - the OKD samples are updated
+# * --ocp - the OCP supported samples are updated
+# * --ocp-unsupported - all OCP samples are updated including the unsupported ones
+
+###########################################
+
+# process the commandline args
+PROCESS_OKD="false"
+PROCESS_OCP_UNSUPPORTED="false"
+if [[ $# -eq 0 ]]; then
+  PROCESS_OCP="true"
+else
+  PROCESS_OCP="false"
+fi
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+  --okd)
+    PROCESS_OKD="true"
+    ;;
+  --ocp)
+    PROCESS_OCP="true"
+    ;;
+  --ocp-unsupported)
+    PROCESS_OCP="true"
+    PROCESS_OCP_UNSUPPORTED="true"
+    ;;
+  esac
+  shift
+done
+
+# set up variables
+OCP_SUPPORTED_SAMPLES="ruby python nodejs perl php httpd nginx eap java webserver dotnet golang rails"
+OCP_ARCHS="ocp-x86_64 ocp-aarch64 ocp-ppc64le ocp-s390x"
+OKD_ARCHS="okd-x86_64"
+ALL_ARCHS="$OCP_ARCHS $OKD_ARCHS"
+if $PROCESS_OKD; then
+  if $PROCESS_OCP; then
+    SUPPORTED_ARCHS=$ALL_ARCHS
+  else
+    SUPPORTED_ARCHS=$OKD_ARCHS
+  fi
+else
+  SUPPORTED_ARCHS=$OCP_ARCHS
+fi
+
+# helper functions
+function reset_directory() {
+  git checkout HEAD -- "$1"
+  # remove any changes from the working tree in this directory that checkout left behind
+  git stash -a -- "$1"
+  git stash drop
+}
+
+function reset_ocp_unsupported() {
+  for d in $(ls); do
+    if [[ "${OCP_SUPPORTED_SAMPLES}" != *"${d}"* ]]; then
```

Comment:
might this reset the okd supported ones as well ?

Reply:
This will just keep the changes only to the samples that are supported in OCP, regardless of the architecture. In another words, this function doesn't care about the architecture, it just resets the unsupported samples in whatever arch it happens to find itself in.

Reply:
right it's minor thing - but it seems that you run this function also for the okd folder at line 129, if there is a bug in the logic you end up resetting those as well. maybe it's an edge case, but you could explicitly exclude okd folders or include only the ocp unsupported folder in the loop which calls this function. 

Anyway feel free to ignore my comment, as said it's not a big issue.